osbrain/tests/test_proxy.py

```diff
@@ -119,6 +122,42 @@ def ping(self):
         InitTimeoutProxy('foo', timeout=1.)
 
 
+def test_agent_proxy_wait_running(nsproxy):
+    """
+    A proxy initialized with running=True should wait for a number of seconds
+    until the agent is running
+    """
+    ap = AgentProcess('agent', nsaddr=nsproxy.addr())
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent', nsproxy.addr())
+    Timer(.5, agent.run).start()
+
+    # Wait long enough for the agent to start running
+    proxy = Proxy('agent', nsproxy.addr(), timeout=1).wait_for_running()
+    assert proxy.ping() == 'pong'
+
+
+def test_agent_proxy_wait_running_timeout(nsproxy):
+    """
+    A proxy initialized with running=True should wait for a number of seconds
+    until the agent is running
+    """
+    ap = AgentProcess('agent', nsaddr=nsproxy.addr())
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent', nsproxy.addr())
+
+    # Wait for too long before starting the agent
+    Timer(3, agent.run).start()
+    with pytest.raises(TimeoutError) as error:
+        Proxy('agent', nsproxy.addr()).wait_for_running(timeout=.5)
+
+    assert "Timed out" in str(error.value)
```

Comment:
To simplify this test, I think you could remove the `nsproxy.addr()` from all calls.